tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,200 @@
+				Success("Istio CRDs are present")
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio resource", func() {
+		AfterAll(func() {
+			// Cleanup the operator at the end of the test
+			Eventually(undeployOperator).Should(Succeed(), "Operator failed to be deleted; unexpected error")
+			Success("Operator deployment is deleted")
+		})
+
+		for _, version := range istioVersions {
+			// Note: This var version is needed to avoid the closure of the loop
+			version := version
+
+			Context(fmt.Sprintf("version %s", version), func() {
+				BeforeAll(func() {
+					Expect(kubectl.CreateNamespace(controlPlaneNamespace)).To(Succeed(), "Namespace failed to be created; unexpected error")
+
+					By("Creating Istio CR")
+					Eventually(createIstioCR).
+						WithArguments(version).
+						Should(Succeed(), "Istio CR failed to be created; unexpected error")
+					Success("Istio CR created")
+				})
+
+				AfterAll(func() {
+					By("Cleaning up")
+					Eventually(kubectl.DeleteNamespace).
+						WithArguments(controlPlaneNamespace).
+						Should(Succeed(), "Namespace failed to be deleted; unexpected error")
+
+					Eventually(kubectl.CheckNamespaceExist).
+						WithArguments(controlPlaneNamespace).
+						Should(MatchError(kubectl.ErrNotFound), "Namespace should not exist; unexpected error")
+					Success("Cleanup done")
+				})
+
+				When("the resource is created", func() {
+					It("updates the Istio resource status to Reconcilied and Ready", func() {
+						Eventually(kubectl.GetResourceCondition).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReconcilied), "Istio it's not Reconcilied; unexpected Condition")
+
+						Eventually(kubectl.GetResourceCondition).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReady), "Istio it's not Ready; unexpected Condition")
+
+						Eventually(kubectl.GetPodPhase).
+							WithArguments(controlPlaneNamespace, "app=istiod").
+							Should(Equal("Running"), "Istiod is not running; unexpected pod Phase")
+						Success("Istio resource is Reconcilied and Ready")
+					})
+
+					It("deploys correct istiod image tag according to the version in the Istio CR", func() {
+						// TODO: we need to add a function to get the istio version from the control panel directly
+						// and compare it with the applied version
+						// This is a TODO because actual version.yaml contains for example latest and not the actual version
+						// Posible solution is to add actual version to the version.yaml
+					})
+
+					It("doesn't continuously reconcile the istio resource", func() {
+						Eventually(kubectl.GetPodLogs).
+							WithArguments(controlPlaneNamespace, "app=istiod", 30*time.Second).
+							ShouldNot(ContainSubstring("Reconciliation done"), "Istio Operator is continuously reconciling")
+						Success("Istio Operator stopped reconciling")
+					})
```

Comment:
This should be moved to the end, as checking what the operator has done is more important than checking that it's has stopped doing stuff after it has done what it needs to.